airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -0,0 +1,255 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import time
+import math
+import requests
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+
+# from airbyte_cdk.models import SyncMode
+# from airbyte_cdk.sources.streams.http.auth import HttpAuthenticator
+
+
+class ShopifyStream(HttpStream, ABC):
+    url_base = ""
```

Comment:
this can be `https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/` and then the paths in subclasses can just be the `{self.api_version}....` portion?

Reply:
also I think we can just use the `Authorization` header like described in the [docs](https://shopify.dev/tutorials/authenticate-a-private-app-with-shopify-admin#make-authenticated-requests): 

> If your HTTP client doesn't support basic authentication using this method, then you can provide the credentials in the Authorization header field instead:

Reply:
Basically that's the same, I used, but I didn't use base64 in order to encode the base URL. I'll try to implement this, thanks.